integration-tests/kubernetes-client/src/test/java/io/quarkus/it/kubernetes/client/KubernetesClientSerializationTest.java

```diff
@@ -0,0 +1,144 @@
+package io.quarkus.it.kubernetes.client;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+import java.util.function.Consumer;
+
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.ContainerBuilder;
+import io.fabric8.kubernetes.api.model.ContainerPortBuilder;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.api.model.PodBuilder;
+import io.fabric8.kubernetes.api.model.PodSpecBuilder;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.server.mock.KubernetesServer;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.kubernetes.client.KubernetesTestServer;
+import io.quarkus.test.kubernetes.client.WithKubernetesTestServer;
+import io.restassured.filter.log.LogDetail;
+
+@WithKubernetesTestServer(setup = KubernetesClientSerializationTest.CrudEnvironmentPreparation.class)
+@QuarkusTest
+public class KubernetesClientSerializationTest {
+
+    @KubernetesTestServer
+    private KubernetesServer mockServer;
+
+    @Test
+    @DisplayName("PUT with String body - Should serialize and persist the Pod")
+    void serialization() {
```

Comment:
I'm not sure this is really testing what I was looking for. In the problematic test we start by building a `Pod` instance and how that gets transform into the actual JSON is the problematic area. Is this really being tested here?

Reply:
Ah never mind, it's deserialization it seems